src/ray/gcs/gcs_server/gcs_worker_manager.h

```diff
@@ -73,6 +85,16 @@ class GcsWorkerManager : public rpc::WorkerInfoHandler {
   std::vector<std::function<void(std::shared_ptr<WorkerTableData>)>>
       worker_dead_listeners_;
 
+  /// A list where workers are store as pairs of (WorkerID, Timestamp).
+  /// The workers are sorted according to the timestamp, and the oldest is at the head of the list.
+  /// @note The pair consists of:
+  ///   - first: WorkerID (identifier for the worker)
+  ///   - second: Timestamp (time when the worker was last updated or added)
+  std::list<std::pair<WorkerID, int64_t>> sorted_dead_worker_list_;
```

Comment:
`std::list`'s performance is usually worse than vector due to poor cache locality, I guess list is used here for pop front operation? What about `std::deque`?

I only use `std::list` when insert or removal from the middle of the container is needed.

Reply:
Yeah my point is, you could use 
```
std::deque<std::pair<ActorID, int64_t>>
```
instead of 
```
std::list<std::pair<ActorID, int64_t>>
```
which supports (1) cheap pop front operation; (2) better sorting and overall operation performance

Reply:
I do it